CMakeLists.txt

```diff
@@ -142,6 +142,14 @@ else()
   assert(0 "Unreachable code. Expected optimization level to have been chosen. See Kconfig.zephyr")
 endif()
 
+if(NOT (CONFIG_ARC OR CONFIG_ARM OR CONFIG_X86 OR CONFIG_NIOS2 OR CONFIG_RISCV OR
```

Comment:
This would be quite annoying for out of tree architectures (like we have)

Reply:
Could the arch config's select HAS_ARCH?

Then we could just assert that HAS_ARCH is set.

Reply:
An intermediate symbol like that would certainly solve the problem

Reply:
Changed it to use a helper symbol.

Not sure how common this issue would be tbh, but might be nice for people who don't notice this change (though they'd get errors from assigning `CONFIG_<arch>` symbols).